entity-service-change-event-generator/src/main/java/org/hypertrace/entity/service/change/event/impl/EntityChangeEventGeneratorImpl.java

```diff
@@ -123,6 +125,9 @@ public void sendChangeNotification(
 
   private void sendCreateNotification(RequestContext requestContext, Entity createdEntity) {
     try {
+      this.entityMetricsRegistry
```

Comment:
can we move this to the create code? reporting metrics and change events are different responsibilities, it's unintuitive that the metrics are incremented only as a side effect of sending a change notification.

Reply:
Okay here is my reasoning. Let's forget the name for a sec, what this class is doing is evaluating the change because of an api request and reporting them to downstream service (in the form of events). The problem with implementing this metric in create code is that it's difficult/duplicate code to evaluate what has been updated, created, and delete (specifically in case of upserts, update by query and delete by query). Though, I can move the code of Entity Change evaluation to the create class/package but decided because we are taking this decision based on the change evaluation only.

Reply:
I'm not quite sure I'm following. If this call was right here, vs right before whatever code calls `sendCreateNotification`, does that change the logic required?

My comment was really about responsibility delegation - presumably whatever code is calling to send these notifications already knows what action is being taken, and should be able to deal with this.

Reply:
oh i get it.

> presumably whatever code is calling to send these notifications already knows what action is being taken, and should be able to deal with this.

So this assumption is not correct. For example in the case of upsert, calling class calls for shouldSendNotification and it evaluates whether the upsert resulted in update or create and then send the respective notifications